### ISSUE TITLE
Tags and release notes

### ISSUE THREAD

**Issue by ajwhite** [howdyai/botkit#155]
#### :memo: Tags & Releases :shipit:

This project is rightfully becoming very popular and is the underlying piece behind many Slackbots we see and use today. It's important for developers to keep up with the release track - Botkit contributors are great about handling bugfixes and introducing enhancements. The only problem is that it's **very hard to know what has changed between releases**.

<img width="771" alt="screen shot 2016-03-18 at 11 46 12 pm" src="https://cloud.githubusercontent.com/assets/656630/13896499/a377fb60-ed63-11e5-9d04-65893eb044d0.png">

This thread may be familiar to some -- last week we were jumping on the latest version but didn't have a great way to gloss over the changes we were inheriting.

#### Option 1: Slack

@benbrown always lets users know in Slack that a new version is out with a mention of what's changed (thanks!).

<img width="466" alt="screen shot 2016-03-18 at 11 08 50 pm" src="https://cloud.githubusercontent.com/assets/656630/13896338/6f352832-ed5e-11e5-9e8b-8861b526b9d3.png">

#### Option 2: Check `package.json` followed by commit history review

Periodically check the project's package file for a newer version and then review the commit history to understand what's changed. Sometimes the summaries are self explanatory, however sometimes opening a commit or PR and reviewing the changes are necessary to understand the impact.

<img width="557" alt="screen shot 2016-03-18 at 11 12 56 pm" src="https://cloud.githubusercontent.com/assets/656630/13896351/fa0a3268-ed5e-11e5-973f-c1123b7fbac9.png">

## How we follow other project progress

[microsoft/react-native-code-push](https://github.com/Microsoft/react-native-code-push/releases) provides a good example of Github's Releases section. There's a summary of the release outlining new features, bugfixes, and **breaking changes**. We recognize Botkit does not always have large releases, but a couple bullet points mentioning the changes goes a long way. This way when ready to catch up to the latest version, we can just pop over to the Releases tab and see what's new.

<img width="886" alt="screen shot 2016-03-18 at 11 18 41 pm" src="https://cloud.githubusercontent.com/assets/656630/13896372/c6bfadba-ed5f-11e5-873c-82ededf5202e.png">

## Different release note strategies
- [microsoft/react-native-code-push](https://github.com/Microsoft/react-native-code-push/releases) - Github Releases
- [github/hubot](https://github.com/github/hubot/pull/1035) - tags linking to PR's with release summaries
- [angular/angularjs](https://github.com/angular/angular.js/blob/master/CHANGELOG.md) - CHANGELOG.md

----

TL;DR - Botkit has amazing momentum, and it would be fantastic if release notes were included on tags so the development community can follow along.

**Comment by peterswimm** [comment]
Release Notes are [now live](https://github.com/howdyai/botkit/blob/master/changelog.md)

**Comment by ajwhite** [comment]
Thanks! 👏
